internal/lsp/server_test.go

```diff
@@ -1095,17 +1097,39 @@ func createConnections(
 }
 
 // NewTestLogger returns an io.Writer that logs to the given testing.T.
+// This is helpful as it can be used to have the server log to the test logger
+// in server tests. It is protected from being written to after the test is
+// over.
 func newTestLogger(t *testing.T) io.Writer {
 	t.Helper()
 
-	return &testLogger{t: t}
+	tl := &testLogger{t: t, open: true}
+
+	// using cleanup ensure that no goroutines attempt to write to the logger
+	// after the test has been cleaned up
+	t.Cleanup(func() {
+		tl.mu.Lock()
+		defer tl.mu.Unlock()
+		tl.open = false
+	})
```

Comment:
When the test is over, the logger is closed to stop it accepting more logs and causing a panic when the test is done.